File: tests/fixtures/patterns-positive/abstract_factory/mutB.ts
// POSITIVO (mutación B: Abstract Factory aplanada en 2 cadenas condicionales paralelas).
function createButton(kind: "win" | "mac"): Button {
  if (kind === "win") {
    return new WinButton();
  } else if (kind === "mac") {
    return new MacButton();
  }
  throw new Error("kind desconocido");
}
function createCheckbox(kind: "win" | "mac"): Checkbox {
  if (kind === "win") {
    return new WinCheckbox();
  } else if (kind === "mac") {
    return new MacCheckbox();
  }
  throw new Error("kind desconocido");
}

File: tests/fixtures/patterns-positive/abstract_factory/mutB.py
# POSITIVO (mutación B: Abstract Factory aplanada en 2 cadenas condicionales paralelas).
def create_button(kind):
    if kind == "win":
        return WinButton()
    elif kind == "mac":
        return MacButton()
    raise ValueError("kind desconocido")


def create_checkbox(kind):
    if kind == "win":
        return WinCheckbox()
    elif kind == "mac":
        return MacCheckbox()
    raise ValueError("kind desconocido")
